crates/dojo-lang/src/compiler.rs

```diff
@@ -131,8 +143,37 @@ impl Compiler for DojoCompiler {
         };
 
         let mut compiled_classes: HashMap<String, (Felt, ContractClass)> = HashMap::new();
+        let list_selector =  get_allowed_libfuncs_list(&props, &unit);
 
         for (decl, class) in zip(contracts, classes) {
+            match class.validate_version_compatible(list_selector.clone()) {
+                Ok(()) => {},
+                Err(AllowedLibfuncsError::UnsupportedLibfunc {
+                    invalid_libfunc,
+                    allowed_libfuncs_list_name,
+                }) => {
+                    let contract_name = decl.submodule_id.name(db.upcast_mut());
+                    let diagnostic = formatdoc! {r#"
+                        Contract `{contract_name}` include `{invalid_libfunc}` function that is not allowed in the libfuncs list `{allowed_libfuncs_list_name}`.
```

Comment:
```suggestion
                        Contract `{contract_name}` ({qualified_path}) include `{invalid_libfunc}` function that is not allowed in the default libfuncs list.
```
Let's include the qualified path for better debugging.